File: rvc_service/RVCSDK/examples/CSharp/GetCameraParameter.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RVC_CSharp
{
    class Program
    {

        static void Main()
        {
            #region Step 0 ,Init & Find & Open

            if (false == RVC_CSharp.System.Init())
            {
                Console.WriteLine("Failed to init system.");
                Console.ReadKey();
                return;
            }

            Device device = RVC_CSharp.System.FindDeviceByIndex(0);
            if (false == device.IsValid() || false == device.CheckCanConnected())
            {
                Console.WriteLine("Device Can not be Connected.\nPlease Check Connection.");
                RVC_CSharp.System.Shutdown();
                Console.ReadKey();
                return;
            }
            if(device.IsFirmwareMatch() == false)
            {
                Console.WriteLine("Device firmware mismatch. Please use RVCManager to upgrade the firmware");
                RVC_CSharp.System.Shutdown();
                Console.ReadKey();
                return;
            }
            X1 camera = RVC_CSharp.X1.Create(device, CameraID.CameraID_Left);
            camera.Open();
            if (false == camera.IsValid() || false == camera.IsOpen())
            {
                Console.WriteLine($"Failed to open camera .Error: { System.GetLastErrorMessage() } ");
                RVC_CSharp.System.Shutdown();
                Console.ReadKey();
                return;
            }
            // optional 
            // Print Device Info
            DeviceInfo info = new DeviceInfo();
            device.GetDeviceInfo(ref info);
            info.Print();

            #endregion

            #region Step 1 ,GetCameraParameter
            float[] params_intrinsic = new float[9];
            float[] distortion = new float[5];
            float[] T_extrinsic = new float[16];
            bool ret = camera.GetExtrinsicMatrix(T_extrinsic);
            bool ret2 = camera.GetIntrinsicParameters(params_intrinsic, distortion);

            ret &= ret2;
            if (ret)
            {
                Console.WriteLine("camera Extrinsic Matrix: ");
                for (int i = 0; i < 16; i++)
                {
                    Console.Write(T_extrinsic[i].ToString() + ", ");
                }
                Console.WriteLine("\ncamera intrinsic parameter: ");
                for (int i = 0; i < 9; i++)
                {
                    Console.Write(params_intrinsic[i].ToString() + ", ");
                }
                Console.WriteLine("\ncamera distortion: ");
                for (int i = 0; i < 5; i++)
                {
                    Console.Write(distortion[i].ToString() + ", ");
                }
                Console.WriteLine("\n\n");
            }
            else
            {
                Console.WriteLine(System.GetLastErrorMessage());
                Console.WriteLine("RVC X camera is not valid !!!");
            }
            #endregion

            #region Step 2 ,Close

            camera.Close();
            camera.Destroy();

            RVC_CSharp.System.Shutdown();
            Console.WriteLine("System close.");

            Console.ReadKey();
            #endregion
            return;
        }

    }
}

File: rvc_service/RVCSDK/examples/Python/GetCameraParameter.py
# Copyright (c) RVBUST, Inc - All rights reserved.
import PyRVC as RVC
import os
import numpy as np
import cv2
from Utils.Tools import *

def App():

    # Initialize RVC system.
    RVC.SystemInit()

    # Choose RVC Camera type (USB, GigE or All)
    opt = RVC.SystemListDeviceTypeEnum.All

    # Scan all RVC Camera devices.
    ret, devices = RVC.SystemListDevices(opt)
    print("RVC Camera devices number:%d" % len(devices))

    #  Find whether any RVC Camera is connected or not.
    if len(devices) == 0:
        print("Can not find any RVC Camera!")
        RVC.SystemShutdown()
        return 1
    print("devices size = %d" % len(devices))
    if devices[0].IsFirmwareMatch() == False:
        print("device firmware mismatch, Please use RVCManager to upgrade the firmware")
        RVC.SystemShutdown()
        return 1
    # Create a RVC Camera and choose use left side camera.
    x = RVC.X1.Create(devices[0], RVC.CameraID_Left)

    # Test RVC Camera is valid or not.
    if x.IsValid() == True:
        print("RVC Camera is valid!")
    else:
        print("RVC Camera is not valid!")
        RVC.X1.Destroy(x)
        RVC.SystemShutdown()
        return 1
    #PrintCaptureMode(devices[0])
    
    x.Open()
    
    if x.IsOpen() == False:
        print("Failed to open camera! Please check whether the camera is connected and make sure it is not occupied and supports X1.")
        RVC.X1.Destroy(x)
        RVC.SystemShutdown()
        exit(1)
    

    ret, T_extrinsic = x.GetExtrinsicMatrix()
    if ret == True:
        print("\nExtrinsicMatrix:")
        print(T_extrinsic)

    ret, T_intrinsic_matrix, T_distortion = x.GetIntrinsicParameters()
    if ret == True:
        print("\nIntrinsicParameters:")
        print(T_intrinsic_matrix)
        print("\ndistortion:")
        print(T_distortion)

    # Close RVC Camera.
    x.Close()

    # Destroy RVC Camera.
    RVC.X1.Destroy(x)

    # Shutdown RVC System.
    RVC.SystemShutdown()

    return 0


if __name__ == "__main__":
    App()
